<feat> Organization Page

danielpeterkim opened the pull request:
Organization page using dummy data, and a non-functional google maps embed.

To access go to route organization/:id
The id can be anything currently as they all go to the same place.

danielpeterkim committed:
<feat> organization page

Organization page using dummy data, and a google maps embed. Does not contain image support

danielpeterkim committed:
Update Remove Unused Tabs

danielpeterkim commented:
Nvm, it was just a unused package. My b.

danielpeterkim committed:
Added Picture

danielpeterkim committed:
Prettier Formating

Updated for Prettier Formatting for Build

miguel-merlin commented:
@danielpeterkim For future pull requests, can you include screenshots of the changes you made

miguel-merlin commented on src/components/OrganizationDisplayCard.tsx:
```diff
@@ -0,0 +1,47 @@
+import { Card } from 'antd';
+
+// The Left Column Top Card
+const OrganizationDisplayCard: React.FC = () => {
```
I think `React.FC` is going to get deprecated at some point. Just do `const OrganizationDisplayCard = () => {}`

miguel-merlin commented on src/components/OrganizationDisplayCard.tsx:
```diff
@@ -0,0 +1,47 @@
+    <Card
+      title={<span style={{ fontSize: '2rem' }}>A Wider Circle</span>}
+      bordered
+      style={{
```
We usually prefer using tailwind for styling. Can you change the `style` prop for `className= ...`

miguel-merlin commented on src/components/OrganizationDisplayCard.tsx:
```diff
@@ -0,0 +1,47 @@
+        boxShadow: '0 4px 8px rgba(0,0,0,0.2)',
+      }}
+    >
+      <div style={{ fontSize: '1rem', color: '#333' }}>
```
Same comment here about tailwind

miguel-merlin commented on src/components/OrganizationDisplayCard.tsx:
```diff
@@ -0,0 +1,47 @@
+import React from 'react';
+import { Card } from 'antd';
```
Could you maybe add an interface prop to this component?
```ts
interface OrganizationDisplayCard {
   organization: Organization
}
```
You will need to import the type

miguel-merlin commented on src/components/OrganizationTabs.tsx:
```diff
@@ -0,0 +1,194 @@
+
+//   Example "Services" data array.
+//   In a real app, you'll fetch this from a backend and store it in state or props.
+const servicesData = [
```
Maybe this move this array to `data/services.ts` and import it from there. It will be cleaner

danielpeterkim replied:
that makes sense, will do

miguel-merlin commented on src/components/OrganizationTabs.tsx:
```diff
@@ -0,0 +1,194 @@
+//  you can create separate components as shown here.
+
+// Render a "Service" item
+const ServiceCard: React.FC<{
```
`React.FC`

miguel-merlin commented on src/components/OrganizationTabs.tsx:
```diff
@@ -0,0 +1,194 @@
+
+// Render a "Service" item
+const ServiceCard: React.FC<{
+  title: string;
```
I think for this, it will be just easier to take in a `Service` object. Can you change the props of the component to
```ts
interface ServiceCardProps {
   service: Service
}
```
You will have to import the `Service` interface

miguel-merlin commented on src/components/OrganizationTabs.tsx:
```diff
@@ -0,0 +1,194 @@
+      <p>
+        <strong>Address:</strong> {address}
+      </p>
+      {/* Example "chips" for categories */}
```
I don't think we need these comments

miguel-merlin commented on src/components/OrganizationTabs.tsx:
```diff
@@ -0,0 +1,194 @@
+};
+
+// Render a "Location" item
+const LocationCard: React.FC<{
```
Can you add this card in another file? `LocationCard.tsx`

miguel-merlin commented on src/pages/OrganizationLayout.tsx:
```diff
@@ -0,0 +1,92 @@
+    <>
+      <Navbar />
+      {/* PLACE HOLDER SEARCH BAR*/}
+      {/* TO DO: Add search bar that links back to "Organizations" page with search query */}
```
Is the todo going to be addressed in another PR?

danielpeterkim replied:
Yes, I didn't really understand the functions in the Navbar of the Organization page at the time, but I would like to revisit it in a new PR.

miguel-merlin commented on src/pages/OrganizationLayout.tsx:
```diff
@@ -0,0 +1,92 @@
+        <Content style={{ margin: '2rem' }}>
+          <div style={{ display: 'flex', gap: '2rem', flexWrap: 'wrap' }}>
+            {/* Left Column: Organization Display Card */}
+            <div style={{ flex: '1 1 400px', minWidth: '350px' }}>
```
tailwind

danielpeterkim replied:
Yeah, will do. Saw the use of style for the drop shadows and totally forgot about tailwind. :P

miguel-merlin commented on src/pages/OrganizationLayout.tsx:
```diff
@@ -0,0 +1,92 @@
+                </div>
+              </Card>
+              {/* Google Map Embed */}
+              <iframe
```
We have a component? Why not use it here?

danielpeterkim replied:
Yeah, map was empty at that time but definitely could have just put it there, whoopsy.

miguel-merlin reviewed (changes requested):
Good work! I left some small comments!

danielpeterkim committed:
Rewrote styling to Tailwind, Added Map to OrganzitionLayout, and Other

Addressed and changed various styling inconsistencies, switching all CSS to tailwind. Created Seperate component file for location card, and moved dummy data to Service Data,

danielpeterkim committed:
Created Actual Dummy Data

Created life-like dummy data, allowing me to change all the pages to take in actual data objects like Services and Organizations.

danielpeterkim commented:
![image](https://github.com/user-attachments/assets/1fadadfd-40a4-4571-9f8e-5c1eea7554e0)

Switched all Organization Page styling to Tailwind and Added Map Embed

![image](https://github.com/user-attachments/assets/482a8d9c-91a3-46fb-8fb7-6c94b9e2bf63)

Created a DummyData.tsx, which has Organization, Service, and Location Objects (use responsibly as some field were removed for convenience)

![image](https://github.com/user-attachments/assets/6e2a6a7a-d8f4-4ce0-815a-34fcc60ae343)
Created a OrganizationLayoutServiceCard.tsx for the Service cards in the Organization Page (they display more information than ServiceCard.tsx)

![image](https://github.com/user-attachments/assets/1da74652-7a6b-4318-962d-541219ae48ca)

OrganizationTabs.tsx functionally take in Service and Location Data to create tabs, although the LocationCard input data is preprocessed.

danielpeterkim committed:
Fixed Some DummyData Issues

Various data types were inputted incorrectly, I fixed them

danielpeterkim commented:
I can't seem to find why it isn't building. MapProps isn't used, while Maps itself doesn't take in any arguments? Not sure, definite blocker

miguel-merlin commented:
@danielpeterkim We can discuss the build error during the sprint

danielpeterkim committed:
Merge branch 'main' into orgin/feat/org-layout

danielpeterkim committed:
Updating Map Input

Pull from orgin, updating code and adding serviceData to Map component

danielpeterkim commented:
![image](https://github.com/user-attachments/assets/f0b14813-98aa-449b-99fe-30f58f74f320)

Updated from orgin, added serviceData to Map component

miguel-merlin commented on src/App.tsx:
```diff
@@ -18,6 +19,10 @@ const App: React.FC = () => {
           element={<ProtectedRoute element={<Organizations />} />}
         />
+        <Route
+          path="/organizations/:id"
```
`path="/organization/:id"`
